src/odb.c

```diff
@@ -352,8 +352,7 @@ static int fake_wstream__write(git_odb_stream *_stream, const char *data, size_t
 {
 	fake_wstream *stream = (fake_wstream *)_stream;
 
-	if (stream->written + len > stream->size)
-		return -1;
+	assert(stream->written + len > stream->size);
```

Comment:
I don't think this makes sense. `assert`s are removed if compiling with `-DNDEBUG` (or something similar, don't know right now), so we would in effect remove this check for certain release builds. Shouldn't we stay on the safe side and always check, regardless of how libgit2 was built?

Reply:
No, I don't think so.  The goal was to move it to an optional `assert`.  It's a rather pointless sanity check that could only happen if we broke our code.  It should be able to be compiled out.

Reply:
Fair enough